docs/datastorage/config.md

```diff
@@ -0,0 +1,48 @@
+The Config System
+=====================
+
+Sometimes you want to provide some configurable options to your mod.
+You want to save configuration to file and have an ingame gui to change it.
+Then you should use the config system.
+This is an example of how to use it.
+
+```Java
+public class MyConfig {
+  public static final MyConfig INSTANCE;
+  public static final ForgeConfigSpec SPEC;
+
+  // There are other value types
+  public final ForgeConfigSpec.BooleanValue setting;
+    
+  public MyConfig(ForgeConfigSpec.Builder builder) {
+    setting = builder
+      .comment("This is comment")
+      .define("setting", false);
+    }
+    
+  public static boolean getSetting() {
+      return INSTANCE.setting.get();
```

Comment:
You might want to have a section on performance and the advantages of “baking” your config values. While baking you can also perform additional operations, for example turning a list of strings into a set of BlockStates if that’s what the config entry is meant to represent.

Reply:
What do you mean "bake"? Do you mean cache the config value?
Is there an event or callback so that I can update the cache when config updates?
(Actually, I am not familiar with the config system.)

Reply:
Yes, the event. [Here’s](https://cadiboo.github.io/tutorials/1.15.1/forge/3.3-config/) what I’ve got so far (very bare bones, I haven’t given this tutorial as much attention as I usually do), feel free to plagiarise. I’ve found it hard to explain the system and have also had to resort to using examples. I’ll go and change my constant name from “CLIENT” to “INSTANCE”, it’s a much better name. I’ll also get you an example of the transformation I mentioned in a sec

Reply:
It seems that your tutorial is better than this pr. I think that the config doc should be contributed by you-_-

Reply:
Uh oh that’s a lot of responsibility. I’ll do it soon:tm:. When I do, do you mind if I steal from your PR?